### PR TITLE
Complete Deployment distribution

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

* Deployment distribution is simplified and refactored
* processor sends deployment to specific partitions (DISTRIBUTE command)
     * this allows to react on responses from specific partitions
* on response of a partition a COMPLETE DeploymentDistribution is
   written
* this causes the chain of Complete->Completed (eventually
  FULLY_DISTRIBUTED)
* distribution logic is encapsulated in new behavior which is used by
   processor and life cycle listener
 * Life cycle listener triggers redistributing after recovery
 * new integration test for multi partition deployment and recovery
 * test fails without deployment redistributor
 * CREATE deployment processor is renamed
 * removed unused code (column family, some classes etc.)

## Related issues

related #6173 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 66a5b49dfae804faeb860e259bbab45c6ce707c2
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

File: qa/integration-tests/src/test/java/io/zeebe/broker/it/startup/MultiPartitionRecoveryTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  protected void restartBroker(final int nodeId) {
+    clusteringRule.restartBroker(nodeId);
+  }
+
+  private void deploy(final BpmnModelInstance workflowTwoTasks, final String s) {
+    final DeploymentEvent deploymentEvent =
+        clientRule
+            .getClient()
+            .newDeployCommand()
+            .addWorkflowModel(workflowTwoTasks, s)
```
Comment (saig0) [inline qa/integration-tests/src/test/java/io/zeebe/broker/it/startup/MultiPartitionRecoveryTest.java:125]: Correct the parameter name.

```suggestion
  private void deploy(final BpmnModelInstance workflow, final String s) {
    final DeploymentEvent deploymentEvent =
        clientRule
            .getClient()
            .newDeployCommand()
            .addWorkflowModel(workflow, s)
```

File: qa/integration-tests/src/test/java/io/zeebe/broker/it/startup/MultiPartitionRecoveryTest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  public String name;
+
+  public final ClusteringRule clusteringRule = new ClusteringRule();
+  public final GrpcClientRule clientRule = new GrpcClientRule(clusteringRule);
+  @Rule public RuleChain ruleChain = RuleChain.outerRule(clusteringRule).around(clientRule);
+  @Rule public ExpectedException exception = ExpectedException.none();
+  @Rule public Timeout timeout = new Timeout(120, TimeUnit.SECONDS);
+
+  @Parameters(name = "{index}: {1}")
+  public static Object[][] reprocessingTriggers() {
```
Comment (saig0) [inline qa/integration-tests/src/test/java/io/zeebe/broker/it/startup/MultiPartitionRecoveryTest.java:54]: We should change the name. Eventually, there will be no reprocessing anymore :wink:  
We could rename it to `restartAction`. The parameter decides which node should be restarted.

**Review by saig0 — APPROVED** [review]
Looks good :+1: 

Please have a look at the minor suggestions before merging :cake:

**Pull request merged**